### ISSUE TITLE
Error building private extension

### ISSUE THREAD

**Issue by rudygodoy** [GENERALBYTESCOM/batm_public#364]
Hello, because my extension would be very tailored I'm starting it as a private module, so it's separate from `server_extensions_extra` and goes under another directory on the project's top level. 

To begin, essentially, I've just copied `server_extensions_extra` rest example to my module, so I have: `com.generalbytes.batm.server.extensions.cajero.rest`. (I know It can be under a different namespace. Not to worry about it now). I've tweaked root's `gradle.settings` and updated the namespaces and configuration accordingly on my new module. 

Now, I get this weird error when building either the module or the whole project:

```
/batm/batm_public/server_extensions_cajero/src/main/java/com/generalbytes/batm/server/extensions/cajero/rest/RESTServiceCajero.java:112: error: cannot find symbol
            return RESTCajeroExtension.getExtensionContext().calculateCryptoAmounts(serialNumber, Arrays.<String>asList(new String[]{cryptoCurrency}), new BigDecimal(fiatAmount), fiatCurrency, IExtensionContext.DIRECTION_BUY_CRYPTO, null, null);
                                                                                                                                                                                                 ^
  symbol:   variable IExtensionContext
  location: class RESTServiceCajero
```

From this module Gradle's configuration basically I've kept almost everything from `server_extensions_extra` but things related to crypto coins that I'm not going to use as dependencies. Is there anything else where I should look or tweak?

p.s.: I'm aware that later I'll have to figure out how to have my own `batm-extensions.xml` for deployment.

Another related question, would it be a good idea to use the `api` and `implementation` configuration for dependencies on `server_extensions_api` from extra and my private module? 

Thanks!

**Comment by generalbytes** [comment]
My guess is that IExtensionContext is not declared in imports in your class RESTServiceCajero.java.
Yes you need to use api/implementation dependency to project :server_extensions_api from your private module.

**Comment by rudygodoy** [comment]
Hello, the import is included properly and the project builds fine when I remove my module from `settings.gradle`. I'm currently debugging the build process using the Gradle guide for multiple projects builds. Will keep you posted.

```
import com.generalbytes.batm.server.extensions.AbstractExtension;
import com.generalbytes.batm.server.extensions.IExtensionContext;
import com.generalbytes.batm.server.extensions.IRestService;
```

**Comment by rudygodoy** [comment]
I've tried manually compiling those files with `javac`, using the `server_extensions_api.jar` from their own `build/libs` in the classpath along with the other dependencies, and got the same problem. So, I've checked if the class `IExtensionContext` and `ITransactionCashbackInfo` are present and of course they are.

```
% jar -tf /batm/batm_public/server_extensions_api/build/libs/batm_server_extensions_api.jar | grep IExtensionContext
com/generalbytes/batm/server/extensions/IExtensionContext.class
com/generalbytes/batm/server/extensions/IExtensionContext$EmbeddedEmailImage.class
```

Same for `com/generalbytes/batm/server/extensions/ITransactionCashbackInfo.class`. Now, I'm really puzzled.
